RecoLocalCalo/EcalRecAlgos/src/PulseChiSqSNNLS.cc

```diff
@@ -410,8 +411,8 @@ bool PulseChiSqSNNLS::NNLS() {
     
     //adaptive convergence threshold to avoid infinite loops but still
     //ensure best value is used
-    if (iter%50==0) {
-      threshold *= 10.;
+    if (iter % 16 == 0) {
+      threshold *= 2;
```

Comment:
For this case of pathological events it sounds like we don't care how it gets to converge in this case and this update is OK.